### PR TITLE
refactoring(editor): Update editor packages

### PR DESCRIPTION
**Pull request by xrutayisire**
## Context

- Completes DT-1461

## The Solution

- Update editor packages
- Update usage of ErrorBoundary
- Wrap SharedSliceEditor in a Suspense and ErrorBoundary
- Add status in editorConfig properties

## Impact / Dependencies

- Fix the upload button problem from DT-1527

## Checklist before requesting a review
- [x] I hereby declare my code ready for review.
- [ ] If it is a critical feature, I have added tests.
- [ ] The CI is successful.
- [ ] If there could backward compatibility issues, it has been discussed and planned.

### CODE REVIEW

File: packages/slice-machine/components/Simulator/index.tsx [hunk 1]
```diff
@@ -301,21 +311,37 @@ const Simulator: ComponentWithSliceProps = ({ slice, variation }) => {
...
-                }}
-                sharedSlice={sharedSlice}
-              />
+              {/* 
+                The editor warn us it's recommended to wrap  SharedSliceEditor in a Suspense and an ErrorBoundary.
+                Warning: It is not recommended to use editor-support->Suspense without a editor-ui->ErrorBoundary above it 
+              */}
+              <ErrorBoundary
+                renderError={() => (
+                  <DefaultErrorMessage
```
Comment (bapmrl) [inline packages/slice-machine/components/Simulator/index.tsx:320]: ❓ Have we tested the `DefaultErrorMessage` fits nicely in the parent container/pane?

File: packages/slice-machine/src/features/customTypes/customTypesBuilder/PageSnippetDialog/PageSnippetDialog.tsx [hunk 2]
```diff
@@ -78,9 +79,12 @@ export const PageSnippetDialog: FC<PageSnippetDialogProps> = ({ model }) => {
   return (
     <div>
       <ErrorBoundary
-        title="Request failed"
-        description="An error occurred while fetching page types snippets."
-        renderError={() => null}
+        renderError={() => (
+          <DefaultErrorMessage
```
Comment (bapmrl) [inline packages/slice-machine/src/features/customTypes/customTypesBuilder/PageSnippetDialog/PageSnippetDialog.tsx]: ❓ Does it fit in the dialog? Before, we were apparently rendering `null`.

Comment (xrutayisire) [reply packages/slice-machine/src/features/customTypes/customTypesBuilder/PageSnippetDialog/PageSnippetDialog.tsx]: Oh good one, I copy-paste the title and description, but it was returning null. I will just set an empty ErrorBoundary like you back button then 👍

File: packages/slice-machine/components/Simulator/index.tsx (hunk 1 above)
Comment (xrutayisire) [reply packages/slice-machine/components/Simulator/index.tsx:320]: I'm not even sure what are the use cases for us to catch that here... Do you think we should just wrap without message, btw?

**Review by bapmrl — APPROVED** [review]

**Pull request merged**